docker/compute/parflow/Dockerfile

```diff
@@ -0,0 +1,81 @@
+# Docker file for Parflow
+
+ARG BASE_IMAGE=ubuntu:20.04
+
+FROM ${BASE_IMAGE} as build
+RUN apt-get update && \
+    DEBIAN_FRONTEND=noninteractive \
+    apt-get install -y --no-install-recommends \
+        git \
+        wget \
+        ca-certificates \
+        build-essential \
+        gfortran \
+        libopenblas-dev \
+        liblapack-dev \
+        libopenmpi-dev \
+        openmpi-bin \
+        tcl-dev \
+        tk-dev \
+        cmake && \
+    rm -rf /var/lib/apt/lists/*
+
+# ParFlow environment variables
+ENV HYPRE_DIR=/hypre
+ENV PARFLOW_DIR=/parflow
+
+# Install hypre
+RUN wget -q https://github.com/hypre-space/hypre/archive/v2.18.2.tar.gz && \
+    tar xf v2.18.2.tar.gz && \
+    cd hypre-2.18.2/src && \
+    ./configure --prefix=/hypre --with-MPI && \
+    make install && \
+    cd / && \
+    rm -fr hypre-2.18.2 v2.18.2.tar.gz
+
+# Install parflow
+RUN git clone -b master --single-branch https://github.com/parflow/parflow.git /parflow-tmp
+
+RUN mkdir -p /build && \
+    cd /build && \
+    cmake ../parflow-tmp \
+       -DPARFLOW_AMPS_LAYER=mpi1 \
+       -DPARFLOW_AMPS_SEQUENTIAL_IO=TRUE \
+       -DHYPRE_ROOT=/hypre \
+       -DPARFLOW_ENABLE_TIMING=TRUE \
+       -DPARFLOW_HAVE_CLM=TRUE \
+       -DCMAKE_INSTALL_PREFIX=/parflow && \
+    make install && \
+    cd / && \
+    rm -fr /parflow-tmp /build
+
+FROM ${BASE_IMAGE}
+LABEL maintainer="patrick.avery@kitware.com" \
+      version="1.0" \
+      parflow.version="master"
+
+COPY --from=build /parflow /parflow
```

Comment:
Don't you need anything else from the build? Such as `/hypre`?
Garrett may have some input too.

Reply:
So, I took a look inside `/hypre`, and it only appears to contain header files and a static library. I don't think it has any runtime dependencies like shared libraries, so I don't think it's needed at runtime.

I can run this image on their test examples, by the way, and it works for the ones I tried. See [here](https://github.com/Kitware/hpccloud-services/pull/14/files#diff-edc4fb9ca653e11c18b1f715d524018cR13) for an example.